sdk/core/Azure.Core.Experimental/src/Primitives/BinaryData.cs

```diff
@@ -258,21 +315,21 @@ public Stream ToStream()
         /// <param name="data"></param>
         public static implicit operator ReadOnlyMemory<byte>(
             BinaryData data) =>
-            data.AsBytes();
+            data.Bytes;
 
         /// <inheritdoc />
         [EditorBrowsable(EditorBrowsableState.Never)]
         public override bool Equals(object? obj)
```

Comment:
Should we call out here that we are doing something like reference equality rather than value equality, similar to what we have in the comments for `ReadOnlyMemory<T>`'s `.Equals` method?  I'm not sure if customers would expect that this would do value equality or not.